src/pages/Agenda/Agenda.tsx

```diff
@@ -1,13 +1,22 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import "./Agenda.scss";
 import AgendaItem from "./AgendaItem/AgendaItem";
 import TabPanel from "../../components/TabPanel/TabPanel";
 import TabPanelItem from "../../components/TabPanel/TabPanelItem/TabPanelItem";
 import NavigationFooter from "../../components/NavigationFooter/NavigationFooter";
 import FilterIcon from "../../components/icons/FilterIcon/FilterIcon";
 import Categories from "../Categories/Categories";
+import { Session } from "../../types/session";
+import { shortenTitle, dateToTime } from "../../utils/helpers";
+import { STAGES_MAP, DAYS_MAP, CATEGORY_FILTERS } from "../../utils/constants";
 
-const Agenda: React.FC = () => {
+interface AgendaProps {
+  sessions: Session[];
+}
+
+const Agenda: React.FC<AgendaProps> = ({ sessions }) => {
+  const [agendaItemsByDay, setAgendaItemsByDay] = useState<JSX.Element[][]>([]);
+  const [activeAgendaItems, setActiveAgendaItems] = useState<JSX.Element[]>([]);
```

Comment:
IIRC it is not recommended to use JSXElements in the state

Reply:
fixed